pennylane/ops/op_math/controlled_class.py

```diff
@@ -0,0 +1,380 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines the symbolic operation that indicates the control of an operator.
+"""
+
+import warnings
+
+import numpy as np
+from scipy import sparse
+
+import pennylane as qml
+from pennylane import math as qmlmath
+from pennylane import operation
+from pennylane.wires import Wires
+
+from .symbolicop import SymbolicOp
+
+# pylint: disable=too-many-arguments, too-many-public-methods
+class Controlled(SymbolicOp):
+    """Symbolic operator denoting a controlled operator.
+
+    Args:
+        base (~.operation.Operator): the operator that is controlled
+        control_wires (Any): The wires to control on.
+
+    Keyword Args:
+        control_values (Iterable[Bool]): The values to control on. Must be the same
+            length as ``control_wires``. Defaults to ``True`` for all control wires.
+        work_wires (Any): Any auxiliary wires that can be used in the decomposition
```

Comment:
What are the `work_wires`? 😳

Reply:
Currently, `MultiControlledX` has a decomposition process where it can take advantage of additional wires.  If we ever get around to implementing the decomposition for a general n-controlled unitary, we will also take advantage of any extra wires.

These wires are unaffected intermediaries.